### PR TITLE
[Merged by Bors] - chore(Order/InitialSeg): fix simp lemmas

### PR DESCRIPTION
**Pull request by vihdzp**
See [Zulip](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/simp.20and.20subrel).

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/InitialSeg.lean [hunk 1]
```diff
@@ -401,38 +401,30 @@ theorem top_rel_top {r : α → α → Prop} {s : β → β → Prop} {t : γ 
 @[deprecated top_rel_top (since := "2024-10-10")]
 alias topLTTop := top_rel_top
 
-/-- Any element of a well order yields a principal segment -/
-def ofElement {α : Type*} (r : α → α → Prop) (a : α) : Subrel r { b | r b a } ≺i r :=
+/-- Any element of a well order yields a principal segment. -/
+-- The explicit typing is required in order for `simp` to work properly.
```
Comment (jcommelin) [inline Mathlib/Order/InitialSeg.lean:405]: Did you mean `simps`?
```suggestion
-- The explicit typing is required in order for `simps` to work properly.
```

File: Mathlib/Order/InitialSeg.lean [hunk 2]
```diff
@@ -401,38 +401,30 @@ theorem top_rel_top {r : α → α → Prop} {s : β → β → Prop} {t : γ 
...
-theorem ofElement_apply {α : Type*} (r : α → α → Prop) (a : α) (b) : ofElement r a b = b.1 :=
-  rfl
-
 @[simp]
-theorem ofElement_top {α : Type*} (r : α → α → Prop) (a : α) : (ofElement r a).top = a :=
+theorem ofElement_apply {α : Type*} (r : α → α → Prop) (a : α) (b) : ofElement r a b = b.1 :=
   rfl
 
 /-- For any principal segment `r ≺i s`, there is a `Subrel` of `s` order isomorphic to `r`. -/
+-- The explicit typing is required in order for `simp` to work properly.
```
Comment (jcommelin) [inline Mathlib/Order/InitialSeg.lean:416]:
```suggestion
-- The explicit typing is required in order for `simps` to work properly.
```

**Review by jcommelin — COMMENTED** [review]
bors d+

File: Mathlib/Order/InitialSeg.lean (hunk 1 above)
Comment (vihdzp) [reply Mathlib/Order/InitialSeg.lean:405]: No, the issue arises with any simp lemmas involving this definition.

**Pull request closed without merging**